### PR TITLE
Feature - integrate cloud role instance into application enricher

### LINKED ISSUE #110: Do we expect people to provide Machine Name or do we help them?
Our Application Insights sink automatically assigned `CloudContext.RoleInstance` based on either pod name or machine name ([code](https://github.com/arcus-azure/arcus.observability/blob/master/src/Arcus.Observability.Telemetry.Serilog.Sinks.ApplicationInsights/Converters/CloudContextConverter.cs#L27)).

However, for Machine Name you have to know to use [Serilog.Enrichers.Environment](https://github.com/serilog/serilog-enrichers-environment) but have to know that we depend on this.

So I'm wonderin:
- Should we just document this and add it to our templates, or
- Should we base our [`ApplicationEnricher`](https://github.com/arcus-azure/arcus.observability/blob/master/src/Arcus.Observability.Telemetry.Serilog.Enrichers/ApplicationEnricher.cs) on it and extend it?

### PR DESCRIPTION
**Pull request by stijnmoreels**
Started something but will need some help with code location, naming and setup. Thanks.

Closes #110

### CODE REVIEW

File: docs/preview/features/telemetry-enrichment.md [hunk 1]
```diff
@@ -36,7 +36,17 @@ ILogger logger = new LoggerConfiguration()
...
 
-logger.Information("This event will be enriched with the application component's name");
+logger.Information("This event will be enriched with the application component's name and by default the environment's machine name");
+```
+
+Or, alternatively one can choose to use the Kubernetes information.
+
+```csharp
+ILogger logger = new LoggerConfiguration()
+    .Enrich.WithComponentName("My application component", RoleInstance.PodName)
```
Comment (tomkerkhove) [inline docs/preview/features/telemetry-enrichment.md:46]: You should not be choosing this here; we should always use the machine name regardless of Kubernetes.
Our application enricher should just extend the machine name one if you ask me.

If they want to include Kubernetes information they should see our Kubernetes enricher next to this one.

Comment (stijnmoreels) [reply docs/preview/features/telemetry-enrichment.md:46]: Bit then the Kubernetes enricher will not overwrite the existing log property bc it's already present. 🤔

Comment (tomkerkhove) [reply docs/preview/features/telemetry-enrichment.md:46]: You don't have to worry about that, the Kubernetes enricher will just provide the info but the logic is up to our `CloudContextConverter`.

https://github.com/arcus-azure/arcus.observability/blob/147bef3db0c6dd5019b1a68a4f6b7861078751cc/src/Arcus.Observability.Telemetry.Serilog.Sinks.ApplicationInsights/Converters/CloudContextConverter.cs#L31

Comment (stijnmoreels) [reply docs/preview/features/telemetry-enrichment.md]: Updated this.

**Pull request merged**